samplers/aws/xray/internal/rule.go

```diff
@@ -0,0 +1,249 @@
+	}
+
+	// Take from reservoir quota, if quota is available for that second.
+	if r.reservoir.take(now, false, 1.0) {
+		atomic.AddInt64(&r.samplingStatistics.sampledRequests, int64(1))
+		sd.Decision = sdktrace.RecordAndSample
+
+		return sd
+	}
+
+	// using traceIDRatioBased sampler to sample using fixed rate
+	sd = sdktrace.TraceIDRatioBased(r.ruleProperties.FixedRate).ShouldSample(parameters)
+
+	if sd.Decision == sdktrace.RecordAndSample {
+		atomic.AddInt64(&r.samplingStatistics.sampledRequests, int64(1))
+	}
+
+	return sd
+}
+
+// appliesTo performs a matching against rule properties to see
+// if a given rule does match with any of the rule set on AWS X-Ray console.
+func (r *Rule) appliesTo(parameters sdktrace.SamplingParameters, serviceName string, cloudPlatform string) (bool, error) {
+	var httpTarget string
+	var httpURL string
+	var httpHost string
+	var httpMethod string
+	var HTTPURLPathMatcher bool
+
+	if parameters.Attributes != nil {
+		for _, attrs := range parameters.Attributes {
+			if attrs.Key == "http.target" {
+				httpTarget = attrs.Value.AsString()
+			}
+			if attrs.Key == "http.url" {
+				httpURL = attrs.Value.AsString()
+			}
+			if attrs.Key == "http.host" {
+				httpHost = attrs.Value.AsString()
+			}
+			if attrs.Key == "http.method" {
+				httpMethod = attrs.Value.AsString()
+			}
+		}
+	}
+
+	// Attributes and other HTTP span attributes matching.
+	attributeMatcher, err := r.attributeMatching(parameters)
+	if err != nil {
+		return attributeMatcher, err
+	}
+
+	if !attributeMatcher {
+		return attributeMatcher, nil
+	}
+
+	serviceNameMatcher, err := wildcardMatch(r.ruleProperties.ServiceName, serviceName)
+	if err != nil {
+		return serviceNameMatcher, err
+	}
+
+	if !serviceNameMatcher {
+		return serviceNameMatcher, nil
+	}
+
+	serviceTypeMatcher, err := wildcardMatch(r.ruleProperties.ServiceType, cloudPlatform)
+	if err != nil {
+		return serviceTypeMatcher, err
+	}
+
+	if !serviceTypeMatcher {
+		return serviceTypeMatcher, nil
+	}
+
+	HTTPMethodMatcher, err := wildcardMatch(r.ruleProperties.HTTPMethod, httpMethod)
+	if err != nil {
+		return HTTPMethodMatcher, err
+	}
+
+	if !HTTPMethodMatcher {
+		return HTTPMethodMatcher, nil
+	}
+
+	HTTPHostMatcher, err := wildcardMatch(r.ruleProperties.Host, httpHost)
+	if err != nil {
+		return HTTPHostMatcher, err
+	}
+
+	if !HTTPHostMatcher {
+		return HTTPHostMatcher, nil
+	}
+
+	if httpURL != "" {
+		HTTPURLPathMatcher, err = wildcardMatch(r.ruleProperties.URLPath, httpURL)
+		if err != nil {
+			return HTTPURLPathMatcher, err
+		}
+
+		if !HTTPURLPathMatcher {
+			return HTTPURLPathMatcher, nil
+		}
+	} else {
+		HTTPURLPathMatcher, err = wildcardMatch(r.ruleProperties.URLPath, httpTarget)
+		if err != nil {
+			return HTTPURLPathMatcher, err
+		}
+
+		if !HTTPURLPathMatcher {
+			return HTTPURLPathMatcher, nil
+		}
+	}
+
+	return attributeMatcher && serviceNameMatcher && serviceTypeMatcher && HTTPMethodMatcher && HTTPHostMatcher && HTTPURLPathMatcher, nil
```

Comment:
This appears to have been reverted to the original form. Why not keep it returning `true` directly?

Reply:
yeah I think this would be better for readability and aligns more with what we want to do. We want to return true if all of the matchers are true.

Reply:
> yeah I think this would be better for readability and aligns more with what we want to do. We want to return true if all of the matchers are true.

Which we do based on the all the code above? Why check those values twice?

I'm not sure it is more reabable if I'm confused. I expect others to read the redundant code and question if there is something more they are missing.

Reply:
okay fair. I can change it back to `true`